source/modules/validators/validate.py

```diff
@@ -59,7 +59,7 @@ def validate_options(self, options: dict):
             notsupported = ["switches"]
         elif options["subcommand"] == "verify":
             required = ["task", "infra", "stage"]
-            notsupported = ["switches", "commit"]
+            notsupported = ["switches", "commit", "check_mode", "dry_mode"]
```

Comment:
on the other hand.. verifier may also be in ansible... Maybe we shouldn't block that